### PR TITLE
docs: document use of the flaky decorator

### PR DESCRIPTION
**Pull request by emmettbutler**
This pull request documents my thinking about the intended use of the `@flaky` test decorator.

## Checklist

- [x] Change(s) are motivated and described in the PR description
- [x] Testing strategy is described if automated tests are not included in the PR
- [x] Risks are described (performance impact, potential for breakage, maintainability)
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed or label `changelog/no-changelog` is set
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/))
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))
- [x] If this PR changes the public interface, I've notified `@DataDog/apm-tees`.
- [x] If change touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.

## Reviewer Checklist

- [x] Title is accurate
- [x] All changes are related to the pull request's stated goal
- [x] Description motivates each change
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- [x] Testing strategy adequately addresses listed risks
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] Release note makes sense to a user of the library
- [x] Author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

**Review by ZStriker19 — APPROVED** [review]
Love it, thanks for adding

File: docs/contributing-testing.rst [hunk 1]
```diff
@@ -189,3 +189,24 @@ environment object.
...
    ``for h in `riot list --hash-only "^${VENV_NAME}$"`; do rm .riot/requirements/${h}.txt; done; scripts/compile-and-prune-test-requirements``
 4. Commit the resulting changes to the ``.riot`` directory, and open a pull request against the trunk branch.
+
+What do I do when my pull request has failing tests unrelated to my changes?
+----------------------------------------------------------------------------
+
+The test suite is not completely reliable. There are usually some tests that can fail without any of their code paths being
+changed. This slows down development because most tests are required to pass for pull requests to be merged.
+
+The ``tests/utils`` module provides the ``@flaky`` decorator to enable contributors to handle this situation. As a contributor,
```
Comment (erikayasuda) [inline docs/contributing-testing.rst:199]: nit - Could we link to the implementation?
```suggestion
The ``tests/utils`` module provides the [``@flaky`` decorator](https://github.com/DataDog/dd-trace-py/blob/623f2df4de802563a463acc4d3c000dbc742e3d3/tests/utils.py#L1285) to enable contributors to handle this situation. As a contributor,
```

**Review by erikayasuda — APPROVED** [review]
Just one nit, thanks for writing this up!

File: docs/contributing-testing.rst (hunk 1 above)
Comment (erikayasuda) [reply docs/contributing-testing.rst]: @emmettbutler I just previewed the change and it looks strange... Not sure how to make it a normal hyperlink

File: docs/contributing-testing.rst [hunk 2]
```diff
@@ -189,3 +189,24 @@ environment object.
...
    ``for h in `riot list --hash-only "^${VENV_NAME}$"`; do rm .riot/requirements/${h}.txt; done; scripts/compile-and-prune-test-requirements``
 4. Commit the resulting changes to the ``.riot`` directory, and open a pull request against the trunk branch.
+
+What do I do when my pull request has failing tests unrelated to my changes?
+----------------------------------------------------------------------------
+
+The test suite is not completely reliable. There are usually some tests that can fail without any of their code paths being
+changed. This slows down development because most tests are required to pass for pull requests to be merged.
+
+The ``tests/utils`` module provides the [``@flaky`` decorator](https://github.com/DataDog/dd-trace-py/blob/623f2df4de802563a463acc4d3c000dbc742e3d3/tests/utils.py#L1285) to enable contributors to handle this situation. As a contributor,
```
Comment (erikayasuda) [inline docs/contributing-testing.rst:199]: OK I think this should fix the weird link
```suggestion
The ``tests/utils`` module provides the ```@flaky`` decorator <https://github.com/DataDog/dd-trace-py/blob/623f2df4de802563a463acc4d3c000dbc742e3d3/tests/utils.py#L1285>`_ to enable contributors to handle this situation. As a contributor,
```

**Pull request merged**